lib/stm32/g4/adc.c

```diff
@@ -544,10 +544,8 @@ void adc_set_clk_source(uint32_t adc, uint32_t source)
  */
 void adc_set_clk_prescale(uint32_t adc, uint32_t prescale)
 {
-	uint32_t reg32 = ADC_CCR(adc);
-
-	reg32  &= ~(ADC_CCR_PRESC_MASK << ADC_CCR_PRESC_SHIFT);
-	ADC_CCR(adc) = (reg32 | (prescale << ADC_CCR_PRESC_SHIFT));
+	uint32_t reg32 = ((ADC_CCR(adc) & ~ADC_CCR_PRESC_MASK) | prescale);
+	ADC_CCR(adc) = reg32;
```

Comment:
If the change [above][1] is to be withdrawn, this would need to be changed back to the orignal style, or eventually in a similar way to what you did above:
1. sample the register in reg32 (no transformation)
2. apply transformation
3. apply transformatoin
4. write the register back (no transformation)

But this is bikeshedding from me :)

[1]: https://github.com/libopencm3/libopencm3/pull/1458/files#diff-d7c0ebeff6b14b520dee238ddc5e806c2d96728c7d79126fe8895e80322f7883R467

Reply:
Reverted.